lib/commanded/commands/dispatcher.ex

```diff
@@ -103,8 +103,8 @@ defmodule Commanded.Commands.Dispatcher do
         {:exit, {{:nodedown, _node_name}, {GenServer, :call, _}}} ->
           {:error, :remote_node_down}
 
-        {:exit, _reason} ->
-          {:error, :aggregate_execution_failed}
+        {:exit, reason} ->
+          {:error, {:aggregate_execution_failed, reason}}
```

Comment:
```suggestion
          {:error, :aggregate_execution_failed, reason}
```

Returning three elements tuple could be more compatible.


1. the result handles in the next codes can handle these results;
https://github.com/commanded/commanded/blob/7be1943acdbd313707743bcc9142a888ef869dae/lib/commanded/commands/dispatcher.ex#L136-L146
2. by adding new middleware to returning `{:error, {:aggregate_execution_failed, reason}}`, you can get the reason of a `aggregate_execution_failed` error from the dispatching result.

(note: *I'm not a maintainer, just commenting*)

Reply:
I've always found "three-element tuples" to be more trouble.  Often, consuming code wants to bubble up errors.  Two-element tuples make that easy to detect.  Three-element tuples require the surrounding code to add possibly many different matches.  As such, I vastly prefer to have the error "reason" become a tuple (or struct) rather than to change the shape of the return value.

Reply:
Furthermore, assuming I'm reading this correctly, with a three-element tuple, the dispatcher would need to be updated to still respond with the actual reason if you want it in your calling code; otherwise the Pipeline response is dropping the reason and only returning the two-element tuple.